File: src/co/soundwave/repository/song/SongDAO.kt
package co.soundwave.repository.song

import co.soundwave.models.Song
import co.soundwave.repository.BaseDAO

class SongDAO : BaseDAO<SongSQL, Song>() {
    override val sql: SongSQL by lazy { SongSQL() }
    fun getInsertQuery(obj: Song, genreId: Int, albumId: Int): String =
        sql.insertSong(obj.name, obj.track, obj.length, obj.path, genreId, albumId)
}

File: repository/song/SongDAO.py
from models import Song as so
from repository import BaseDAO as bdao
from repository.song import SongSQL as sosql


class SongDAO(bdao.BaseDAO):
    def __init__(self):
        super().__init__()
        self.sql = sosql.SongSQL()

    def get_insert_query(self, song: so.Song, genre_id: int, album_id: int):
        return self.sql.insert_song(
            song.name, song.track, song.length, song.path, genre_id, album_id)
